### ISSUE TITLE
Unable to create an index when routing partition size is set to greater than 1

### ISSUE THREAD

**Issue by mark54g** [elastic/elasticsearch#76989]
**Elasticsearch version** (`bin/elasticsearch --version`):
7.9.3, 7.13.4

**Plugins installed**: []

**JVM version** (`java -version`):

**OS version** (`uname -a` if on a Unix-like system):
Mac OS 10
19.6.0 Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64
Linux of various 

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem,
including (e.g.) index creation, mappings, settings, query etc.  The easier
you make for us to reproduce it, the more likely that somebody will take the
time to look at it.
Create a component template that includes a mapping:

```
PUT _component_template/mv1
 {
   "template": {
     "settings": {
       "index": {
         "number_of_replicas": 0
       }
     },
     "mappings": {
       "properties": {
         "id": {
           "type": "keyword"
         }
       }
     }
   }
 }

```

```
 PUT _index_template/template_w_mappings
 {
   "index_patterns": ["mv1*"],
   "priority": 500,
   "composed_of": ["mv1"],
   "version": 0,
   "_meta": {
    "description": "mark template"
  }
 
 }
```

Try to create an index using `routing_partition_size` greater than one and enable a mapping that has `_routing` required as true.

```
PUT mv1-1
 {
   "settings": {
     "number_of_shards": 10,
     "number_of_routing_shards": 10,
     "routing_partition_size": 2
   },
  "mappings": {
     "_routing": {
       "required": true
    }
   }
 }
```

This results in:

```
{
   "error" : {
     "root_cause" : [
       {
         "type" : "illegal_argument_exception",
         "reason" : "mapping type [_doc] must have routing required for partitioned index [mv1-1]"
       }
     ],
     "type" : "illegal_argument_exception",
     "reason" : "mapping type [_doc] must have routing required for partitioned index [mv1-1]"
   },
   "status" : 400
 }
```

**Provide logs (if relevant)**:

**Comment by wjp719** [comment]
the issue can be solved in pr https://github.com/elastic/elasticsearch/pull/74394

**Comment by arteam** [comment]
Fixed via #74394
